design/automated-airgap-installations.md

```diff
@@ -0,0 +1,77 @@
+# Automated Airgap Installations and Updates
+
+Automated installs in KOTS are limited to online installs only.
+It's not possible to automate an end-to-end installation of an application in airgapped mode.
+By adding automation of airgapped applications, we can also enable a workflow for installing applications to airgapped (existing) clusters.
+
+## Goals
+
+- Automated installs of KOTS (Admin Console) and the application to an existing airgapped cluster
+-
+
+## Non Goals
+
+- Airgap packaging for non-Replicated applications
+- Distribution of the KOTS CLI as part of the application
+- Support for environments without a local registry
+- Support for non-Linux based container runtimes
+
+## Background
+
+See top section.
+
+
+## High-Level Design
+
+This proposal changes the `kots pull` command to download the `.airgap` bundle from the Replicated servers, unless a path is provided on the command line.
+The `--license-file` parameter will be required.
+A local registry will be required, and it must be accessible from the workstation that is orchestrating this process.
+KOTS pull will load the KOTS images into the registry, and write out the application manifests for KOTS (Admin Console).
+These manifests can be applied using `kubectl`, and include the license file that was supplied, in addition to `--config-values` that were optionally supplied.
+A new automation flag will be passed into the automation license file, via an annotation, that the installation should be completed as airgapped.
+The airgap bundle will be pushed to the local registry, and read from the Admin Console automation.
+
```

Comment:
How about a note about `when the --airgap-package flag is provided, no outbound network requests will be made by KOTS. 

*as of this writing, but subject to change*: 

The only requests made will be to 
- the configured kubecontext's apiserver
- the local registry
- kots in-cluster services (via a local kubectl port forward)